source/apiVolontaria/volunteer/views.py

```diff
@@ -228,6 +229,17 @@ class Events(generics.ListCreateAPIView):
     serializer_class = serializers.EventBasicSerializer
     filter_fields = ['volunteers', 'cycle', 'cell']
 
+    def if_post_permitted(self, request, op_type):
+        # Must get the cell to see if the user is a manager or not
+        cell_id = request.data.get('cell_id', None)
+        cell = Cell.objects.filter(pk=cell_id).first()
+
+        is_user_has_perm = self.request.user.has_perm(op_type)
+        is_cell_manager = cell and request.user in cell.managers.all()
+        can_edit = is_user_has_perm or is_cell_manager
+
+        return can_edit
+
```

Comment:
Why this new structure in place of a standard `permissions` function like you've done for other views ?